pkg/clangtool/tooltest/tooltest.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2025 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package tooltest
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/syzkaller/pkg/clangtool"
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/pkg/testutil"
+)
+
+var (
+	FlagBin    = flag.String("bin", "", "path to the clang tool binary to use")
+	FlagUpdate = flag.Bool("update", false, "update golden files")
+)
+
+func TestClangTool[Output any, OutputPtr clangtool.OutputDataPtr[Output]](t *testing.T) {
```

Comment:
Do you plan to reuse it more in the future?
As of now, having this generic function in `pkg/clangtool` only seems to complicate understanding of the `TestClangTool` test.

Reply:
Yes, I plan to use it for the other tool.